src/charts/helpers/formatters.js

```diff
@@ -101,20 +101,33 @@ export function multiFormat(date) {
  *  Format dates when binned by quarter, decade, century
 */
 
-function formatQuarter(val) {
-  val = d3.utcFormat('%m')(val) // convert to integer month (01 - 12)
-  return `Q${Math.floor((parseInt(val, 10) + 3) / 3)}`;
+function formatQuarter(value) {
+  const monthNumber = d3.utcFormat('%m')(value) // convert to integer month (01 - 12)
+  return `Q${Math.floor((parseInt(monthNumber, 10) + 3) / 3)} ${d3.utcFormat('%Y')(value)}`;
 }
 
-export function formatOddDateBin(specifier, val) {
+export function formatOddDateBin(specifier, value) {
   switch (specifier) {
+    // reproducing the old line chart behavior, even if it's wrong
+    case "1w":
+      return `${d3.utcFormat("%b %d")(value)} - ${d3.utcFormat("%b %d,")(d3.utcDay.offset(value, 6))}`
     case "1c":
-      return
+      return `${d3.utcFormat("%Y")(value)} - ${d3.utcFormat("%Y")(d3.utcYear.offset(value, 99))}`
     case "10y":
-      return
+      return `${d3.utcFormat("%Y")(value)} - ${d3.utcFormat("%Y")(d3.utcYear.offset(value, 9))}`
     case "1q":
-      return formatQuarter(val)
+      return formatQuarter(value)
```

Comment:
I would say for consistency sake it would make sense to either have `formatQuarter`'s code in the switch statement OR to make the week, century, and decade formatters their own functions. I vote for the latter so that it's easier to test, but up to you.

Reply:
I inlined the quarter formatter function. I think it's easy enough to test formatOddDateBin as a single export, same as `autoformat` and `getExtractFormatter`.